### ISSUE TITLE
[7.9.9] - Collection relation column broke

### ISSUE THREAD

**Issue by johanvanhelden** [yajra/laravel-datatables#1337]
### Summary of problem or feature request

We have a relation column in a collection table. For example: `relation.attribute`, this worked fine in the old version we had, 7.9.1, but after updating to 7.9.9 we got an error: `ErrorException: Undefined index: relation.attribute in /vendor/yajra/laravel-datatables-oracle/src/Engines/CollectionEngine.php:134`

I can confirm the relation is in the collection, and the attribute exists. And downgrading to 7.9.1 fixes the problem. So there must be a breaking change somewhere between those versions.

### System details
- Operating System: Ubuntu 16.04 LTS
- PHP Version: 7.1.2
- Laravel Version: v5.4.33
- Laravel-Datatables Version: v7.9.9

**Comment by yajra** [comment]
Thanks for reporting. Here is a link to compare the two versions https://github.com/yajra/laravel-datatables/compare/v7.9.1...v7.9.9, I think the recent changes on collection is on multi-column sorting support.

I may not be able to check this yet but will try to as soon as I can. Sorry for the inconvenience. If you can, please do not hesitate to submit a PR. 

BTW, can you please provide some steps to reproduce the issue? Is it on searching, sorting or both?

**Comment by johanvanhelden** [comment]
The error only occurs during the sorting of a relation column in a collection table. Searching works fine.

**Comment by lk77** [comment]
i think you can use alias as a workaround, it's working for me.
```
'client.code as client_code'
```
```
'client.code' => ['data' => 'client_code']
```

**Comment by yajra** [comment]
I just sent a PR for a possible fix. Can you guys please check if it works? Thanks!

**Comment by lk77** [comment]
i don't think so,
i still have undefined index client.name when ordering the name column.

**Comment by yajra** [comment]
@lk77 can you paste some snippet? This is how I tested the PR and it works fine.

```php
        $query = User::with('country')->get();

        return $datatables->collection($query)
                          ->addColumn('action', 'eloquent.tables.users-action')
                          ->make(true);
```

Script:
```js
    $(function () {
        $('#users-table').DataTable({
            serverSide: true,
            processing: true,
            ajax: '/eloquent/collection-data',
            columns: [
                {data: 'id'},
                {data: 'name'},
                {data: 'country.name'},
                {data: 'created_at'},
                {data: 'updated_at'},
                {data: 'action', orderable: false, searchable: false}
            ]
        });
    });
```

**Comment by lk77** [comment]
of course :

```
/**
     * Display ajax response.
     *
     * @return \Illuminate\Http\JsonResponse
     */
    public function ajax()
    {
        return $this->datatables
            ->collection($this->query()->get())
            ->make(true);
    }
```
```
public function query()
    {
        $players = Player::query()->select(['player.id', 'player.name', 'client.name')])->join('client', 'client.id', '=', 'player.client_id');

        return $this->applyScopes($players);
    }
```
```
/**
     * Get columns.
     *
     * @return array
     */
    protected function getColumns()
    {
        return [
            'id' => ['title' => \Lang::trans('index.id'), 'data' => 'id'],
            'name' => ['title' => \Lang::trans('index.name'), 'data' => 'name'],
            'client.name' => ['title' => \Lang::trans('index.client')]
        ];
    }
```

**Comment by yajra** [comment]
I see, when using collection you would have to rely on the actual column name **key** returned in the result array. Hence for this instance, since there will be conflicting name, you have to use alias for the `client.name as client_name` for it to work because you are using join statements. 

But the case would differ if you use eager loading like the way I tested.

**Comment by lk77** [comment]
yes i'm using alias, i removed them from the snippet to keep it simple.

**Comment by yajra** [comment]
Should be fixed on v7.10.1. Thanks!
